interfaces/builtin/download.go

```diff
@@ -0,0 +1,325 @@
+    bus={system,session}
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus={system,session}
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{RequestName,ReleaseName}"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/
+    interface=org.freedesktop.DBus
+    member="GetConnectionAppArmorSecurityContext"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/com/canonical/applications/download/**
+    interface=org.freedesktop.DBus
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/com/canonical/applications/download/**
+    interface=org.freedesktop.DBus.Properties
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+     bus=session
+     path=/com/canonical/applications/download/**
+     interface=com.canonical.applications.Download
+     peer=(label=unconfined),
+
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=downloadCreated
+     peer=(label=unconfined),
+`)
+
+var downloadConnectedSlotAppArmor = []byte(`
+# Allow connected clients to interact with the download manager
+dbus (receive)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=getAllDownloads
+     peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=createDownload
+     peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+     bus=session
+     path=/com/canonical/applications/download/**
+     interface=com.canonical.applications.Download
+     peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=session
+    path=/com/canonical/applications/download/**
+    interface=com.canonical.applications.Download
+    peer=(name=org.freedesktop.DBus, label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=session
+    path=/com/canonical/applications/download/**
+    interface=org.freedesktop.DBus
+    peer=(name=org.freedesktop.DBus, label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=session
+    path=/com/canonical/applications/download/**
+    interface=org.freedesktop.DBus.Properties
+    peer=(name=org.freedesktop.DBus, label=###PLUG_SECURITY_TAGS###),
+
+`)
+
+var downloadConnectedPlugSecComp = []byte(`
+# Description: Can access download manager.
+
+# dbus
+connect
+getsockname
+recvmsg
+send
+sendto
+sendmsg
+socket
+`)
+
+var downloadPermanentSlotSecComp = []byte(`
+# Description: Can act as a download manager.
+
+# dbus
+connect
+getsockname
+recvmsg
+send
+sendto
+sendmsg
+socket
+`)
+
+type DownloadInterface struct{}
+
+func (iface *DownloadInterface) Name() string {
+    return "download"
```

Comment:
I think this is far too generic for an interface name since 'network' allows you to 'download' things. Perhaps 'download-manager'? @niemeyer - what are your thoughts?